prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchemaReader.kt

```diff
@@ -74,12 +106,81 @@ object ConfigSchemaReader : Logging {
         return parentSchema
     }
 
+    private fun processFileSchema(
+        file: File,
+        schemaName: String,
+        ancestry: List<String> = listOf(),
+        schemaClass: Class<out ConfigSchema<out ConfigSchemaElement>>
+    ): ConfigSchema<*> {
+        if (!file.canRead()) throw SchemaException("Cannot read ${file.absolutePath}")
+        val rawSchema = try {
+            readOneYamlSchema(file.inputStream(), schemaClass)
+        } catch (e: Exception) {
+            val msg = "Error while reading schema configuration from file ${file.absolutePath}"
+            logger.error(msg, e)
+            throw SchemaException(msg, e)
+        }
+
+        // set schema name to match the filename
+        rawSchema.name = schemaName
+
+        if (ancestry.contains(rawSchema.name)) {
+            throw HL7ConversionException("Circular reference detected for schema ${rawSchema.name}")
+        }
+        rawSchema.ancestry = ancestry + rawSchema.name!!
+
+        // Process any schema references
+        val rootFolder = file.parent
+        rawSchema.elements.filter { !it.schema.isNullOrBlank() }.forEach { element ->
+            element.schemaRef =
+                readSchemaTreeRelative(element.schema!!, rootFolder, rawSchema.ancestry, schemaClass)
+        }
+        return rawSchema
+    }
+
+    internal fun readSchemaTreeUri(
+        schemaUri: URI,
+        ancestry: List<String> = listOf(),
+        schemaClass: Class<out ConfigSchema<out ConfigSchemaElement>> = ConverterSchema::class.java
+    ): ConfigSchema<*> {
+        val rawSchema = when (schemaUri.scheme) {
+            "file" -> {
+                val file = File(schemaUri)
+                if (!file.canRead()) throw SchemaException("Cannot read ${file.absolutePath}")
+                readOneYamlSchema(file.inputStream(), schemaClass)
+            }
+            "classpath" -> {
+                val input = javaClass.classLoader.getResourceAsStream(schemaUri.path.substring(1))
```

Comment:
Leaving it this way for now, but there might be a cleaner way to do this.  The issue I hit is that 
`URI("classpath:fhirengine/translation/hl7/schema/schema-read-test-07/ORU_R01/patient-identification.yml").path` is `null`

but 
`javaClass.classLoader.getResourceAsStream("/fhirengine/translation/hl7/schema/schema-read-test-07/ORU_R01/patient-identification.yml")` does not return the resource.

Reply:
Tried a couple different options but none were an improvement on yours. Side note, chatGPT came up with the same solution as you :)